programs/locker/src/instructions/v2/cancel_vesting_escrow.rs

```diff
@@ -0,0 +1,155 @@
+        associated_token::token_program = token_program
+    )]
+    pub creator_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        associated_token::mint = token_mint,
+        associated_token::authority = escrow.load()?.recipient,
+        associated_token::token_program = token_program
+    )]
+    pub recipient_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// CHECKED: the creator will receive the rent back
+    #[account(mut)]
+    pub rent_receiver: UncheckedAccount<'info>,
+
+    /// Signer.
+    #[account(mut)]
+    pub signer: Signer<'info>,
+
+    /// Memo program.
+    pub memo_program: Program<'info, Memo>,
+
+    /// Token program.
+    pub token_program: Interface<'info, TokenInterface>,
+}
+
+impl<'info> CancelVestingEscrow<'info> {
+    fn close_escrow_token(&self) -> Result<()> {
+        let escrow = self.escrow.load()?;
+        let escrow_seeds = escrow_seeds!(escrow);
+
+        close_account(CpiContext::new_with_signer(
+            self.token_program.to_account_info(),
+            CloseAccount {
+                account: self.escrow_token.to_account_info(),
+                destination: self.rent_receiver.to_account_info(),
+                authority: self.escrow.to_account_info(),
+            },
+            &[&escrow_seeds[..]],
+        ))?;
+
+        Ok(())
+    }
+}
+
+pub fn handle_cancel_vesting_escrow<'c: 'info, 'info>(
+    mut ctx: Context<'_, '_, 'c, 'info, CancelVestingEscrow<'info>>,
+    remaining_accounts_info: Option<RemainingAccountsInfo>,
+) -> Result<()> {
+    let mut escrow = ctx.accounts.escrow.load_mut()?;
+    let signer = ctx.accounts.signer.key();
+    escrow.validate_cancel_actor(signer)?;
+
+    let current_ts = Clock::get()?.unix_timestamp as u64;
+    let claimable_amount = escrow.get_claimable_amount(current_ts)?;
+    let remaining_amount = ctx
+        .accounts
+        .escrow_token
+        .amount
+        .safe_sub(claimable_amount)?;
+    escrow.cancelled_at = current_ts;
+    require!(escrow.cancelled_at > 0, LockerError::CancelledAtIsZero);
+    drop(escrow);
+
+    // Process remaining accounts
+    let remaining_accounts = if remaining_accounts_info.is_none() {
+        ParsedRemainingAccounts::default()
+    } else {
+        parse_remaining_accounts(
+            &mut ctx.remaining_accounts,
+            &remaining_accounts_info.unwrap().slices,
+            &[AccountsType::TransferHookCancel],
+        )?
+    };
+
+    // Transfer the claimable amount to the recipient
+    transfer_to_user_v2(
+        &ctx.accounts.escrow,
+        &ctx.accounts.token_mint,
+        &ctx.accounts.escrow_token,
+        &ctx.accounts.recipient_token,
+        &ctx.accounts.token_program,
+        Some(MemoTransferContext {
+            memo_program: &ctx.accounts.memo_program,
+            memo: TRANSFER_MEMO_CANCEL_VESTING.as_bytes(),
+        }),
+        claimable_amount,
+        remaining_accounts.transfer_hook_cancel,
```

Comment:
we only have one transfer hook based on token_mint

Reply:
done